metagraph/src/graph/graph_extensions/node_rc.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef __NODE_RC_HPP__
+#define __NODE_RC_HPP__
+
+#include <sdsl/dac_vector.hpp>
+#include <sdsl/rrr_vector.hpp>
+
+#include "graph/representation/base/sequence_graph.hpp"
+#include "common/vectors/bit_vector_adaptive.hpp"
+
+namespace mtg {
+namespace graph {
+
+class DBGSuccinct;
+
+// Maps each node in a PRIMARY-mode DeBruijnGraph to nodes adjacent to its reverse
+// complement. When using the index construct constructor on DBGSuccinct, or when
+// loading such an index, this stores a map from each DBGSuccinct to the BOSS nodes
+// corresponding to the reverse complements of its k-1 prefix and k-1 suffix.
+class NodeRC : public SequenceGraph::GraphExtension {
+  public:
+    using node_index = SequenceGraph::node_index;
+
+    // Use set_graph to set the graph pointer after using the default constructor
+    NodeRC() : graph_(nullptr) {};
+
+    // Construct a NodeRC index
+    NodeRC(const DBGSuccinct &graph);
+
+    void set_graph(const DeBruijnGraph &graph) {
+        assert(!rc_.size() || is_compatible(graph));
+        graph_ = &graph;
+    }
```

Comment:
```suggestion
    // If |construct_index| is false, the index will not be initialized. Method load() must be called.
    NodeRC(const DBGSuccinct &graph, bool construct_index = false);
```

Reply:
I've incorporated this, but with a more accurate comment